qt/python/mantidqt/mantidqt/widgets/sliceviewer/peaksviewer/presenter.py

```diff
@@ -248,12 +248,20 @@ def remove_peaksworkspace(self, name):
         child_presenters = self._child_presenters
         presenter_to_remove = None
         for child in child_presenters:
-            if child.model.peaks_workspace.name() == name:
```

Comment:
Could we implement a similar accessor to retrieve the workspace name on the peaks model? It means we can just hide away the details then.